src/controller/CHIPDeviceController.cpp

```diff
@@ -1034,166 +1061,73 @@ void DeviceCommissioner::OnSessionEstablished()
+    return SendCommand<OperationalCredentialsCluster>(device, request, OnAttestationResponse);
 }
 
-void DeviceCommissioner::OnAttestationResponse(void * context, chip::ByteSpan attestationElements, chip::ByteSpan signature)
+CHIP_ERROR DeviceCommissioner::SendCertificateChainRequestCommand(CommissioneeDeviceProxy * device,
+                                                                  Credentials::CertificateType certificateType)
 {
-    ChipLogProgress(Controller, "Received Attestation Information from the device");
-    DeviceCommissioner * commissioner = reinterpret_cast<DeviceCommissioner *>(context);
-
-    commissioner->mAttestationResponseCallback.Cancel();
-    commissioner->mOnAttestationFailureCallback.Cancel();
+    mCertificateTypeBeingRequested = certificateType;
 
-    commissioner->ValidateAttestationInfo(attestationElements, signature);
+    OperationalCredentials::Commands::CertificateChainRequest::Type request;
+    request.certificateType = CertificateType::kDAC;
+    return SendCommand<OperationalCredentialsCluster>(device, request, OnCertificateChainResponse);
 }
 
-void DeviceCommissioner::OnDeviceAttestationInformationVerification(void * context, AttestationVerificationResult result)
+TRAMPOLINE(AttestationResponse, const OperationalCredentials::Commands::AttestationResponse::DecodableType &);
+void DeviceCommissioner::OnAttestationResponse(const OperationalCredentials::Commands::AttestationResponse::DecodableType & data)
 {
-    DeviceCommissioner * commissioner = reinterpret_cast<DeviceCommissioner *>(context);
-    CHIP_ERROR error                  = CHIP_NO_ERROR;
-
-    if (result != AttestationVerificationResult::kSuccess)
-    {
-        if (result == AttestationVerificationResult::kNotImplemented)
-        {
-            ChipLogError(Controller,
-                         "Failed in verifying 'Attestation Information' command received from the device due to default "
-                         "DeviceAttestationVerifier Class not being overridden by a real implementation.");
-            SuccessOrExit(error = CHIP_ERROR_NOT_IMPLEMENTED);
-        }
-        else
-        {
-            ChipLogError(Controller,
-                         "Failed in verifying 'Attestation Information' command received from the device: err %hu. Look at "
-                         "AttestationVerificationResult enum to understand the errors",
-                         static_cast<uint16_t>(result));
-            // Go look at AttestationVerificationResult enum in src/credentials/DeviceAttestationVerifier.h to understand the
-            // errors.
-            SuccessOrExit(error = CHIP_ERROR_INTERNAL);
-        }
-    }
-
-    ChipLogProgress(Controller, "Successfully validated 'Attestation Information' command received from the device.");
+    StopSessionIf(mState != State::Initialized, CHIP_ERROR_INCORRECT_STATE);
+    StopSessionIf(mDeviceBeingCommissioned == nullptr, CHIP_ERROR_INCORRECT_STATE);
```

Comment:
We didn't use to do these two checks inside `OnAttestationResponse`, right?  Why are we adding them?

Especially given that `ValidateAttestationInfo`, which we are about to call, will do them anyway.

Reply:
I have updated the code to look much closer to the previous state. So those are gone.